### PR TITLE
Log Transforms

### PR DESCRIPTION
**Pull request by AlexRobson**
Adds two forms of log transforms. 

Either:

* LogTransform: sign(x) * log(|x| + 1) 
* InverseHyperbolicSine: log(x + √(x² + 1)).

### CODE REVIEW

File: src/log.jl [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    LogTransform <: Transform
+
+Logarithmically transform the data through: sign(x) * log(|x| + 1).
+
+This allows positive domains elements to be transformed. 
+"""
+struct LogTransform <: Transform end
+cardinality(::LogTransform) = OneToOne()
+
+_logtransform(x) = sign(x) * log(abs(x) + 1)
```
Comment (mzgubic) [inline src/log.jl:11]:
```suggestion
_logtransform(x) = sign(x) * log(abs(x) + one(x))
```

File: src/log.jl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+_invlogtransform(x) = sign(x) * (exp(sign(x) * x) - 1)
```
Comment (mzgubic) [inline src/log.jl:12]:
```suggestion
_invlogtransform(x) = sign(x) * (exp(sign(x) * x) - one(x))
```

File: src/log.jl [hunk 3]
```diff
@@ -0,0 +1,33 @@
+"""
+    LogTransform <: Transform
+
+Logarithmically transform the data through: sign(x) * log(|x| + 1).
+
+This allows positive domains elements to be transformed.
```
Comment (mzgubic) [inline src/log.jl:6]:
```suggestion
This allows transformations of all real numbers, not just positive ones.
```

File: test/log.jl [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            @test cardinality(transform) == OneToOne()
+            @test transform isa Transform
+        end
+
+        @testset "Apply" for (x, y) in ((V1, logV1), (V2, logV2), (M, logM))
+            f = LogTransform()
+            transformed = FeatureTransforms.apply(x, f)
+            @test transformed ≈ y atol=1e-5
+            @test FeatureTransforms.apply(transformed, f; inverse=true) ≈ x atol=1e-5
+        end
```
Comment (mzgubic) [inline test/log.jl:24]: can we add tests for symmetry and going through zero?

File: src/FeatureTransforms.jl [hunk 5]
```diff
@@ -7,7 +7,8 @@ using Tables
 
 export Transform, transform, transform!
 export HoD, LinearCombination, OneHotEncoding, Periodic, Power
-export AbstractScaling, IdentityScaling, MeanStdScaling
+export AbstractScaling, IdentityScaling
+export MeanStdScaling, LogTransform, InverseHyperbolicSine
```
Comment (mzgubic) [inline src/FeatureTransforms.jl:11]:
```suggestion
export AbstractScaling, IdentityScaling, MeanStdScaling
export LogTransform, InverseHyperbolicSine
```
so that linear scalings are together

**Review by mzgubic — APPROVED** [review]
Nice, thanks!

File: test/log.jl (hunk 4 above)
Comment (AlexRobson) [reply test/log.jl:24]: what do you mean by symmetric here? 

For going through zero i assume you mean testing something like: V1 = [1; 2; 3; 4; 0; 6]

Comment (mzgubic) [reply test/log.jl:24]: symmetry: explicitly checking that`log_tsf(x) == log_tsf(-x)`
zero: yes
